File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/Java/MedicDeFamilie/src/familydoctor/repository/AppointmentRepository.java
package familydoctor.repository;

import familydoctor.model.Appointment;

import java.util.stream.Collectors;

public class AppointmentRepository extends AbstractRepository<Appointment> {
    public void add(Appointment appointment) {
        entities.add(appointment);
    }

    public void remove(Appointment appointment) {
        entities.remove(appointment);
    }

    public Appointment getById(String id) {
        return entities.stream().filter(A -> A.getId() == id).collect(Collectors.toList()).get(0);
    }

    public boolean contains(String id) {
        return getById(id) != null;

    }
}

File: Metode avansate de Programare/Teme/familydoctor/Saptamana 8-14/tcp client-server/familydoctor/FamilyDoctorServer/Repositories/AppointmentRepository.cs
using FamilyDoctor.Models;

namespace FamilyDoctor.Repositories
{
    public class AppointmentRepository : Repository<Appointment>
    {
        public AppointmentRepository(string filename)
            : base(filename)
        {
            
        }

        public Appointment GetByID(string id)
        {
            Appointment appointment = Entities.Find(P => P.ID == id);

            if (appointment == null)
                throw new RepositoryException("The specified ID does not exist");

            return appointment;
        }

        public bool Contains(string id)
        {
            return Entities.Find(A => A.ID == id) != null;
        }
    }
}
